src/common/pony/detail/atomics.h

```diff
@@ -119,17 +117,18 @@ namespace ponyint_atomics
 namespace ponyint_atomics
 {
   template <typename T>
-  inline PONY_ABA_PROTECTED(T) big_load(PONY_ABA_PROTECTED(T)* ptr)
+  inline PONY_ABA_PROTECTED_PTR(T) big_load(PONY_ABA_PROTECTED_PTR(T)* ptr)
   {
-    PONY_ABA_PROTECTED(T) ret = {NULL, 0};
+    PONY_ABA_PROTECTED_PTR(T) ret = {NULL, 0};
```

Comment:
It turns out it is also possible in `pool_block_push/pull` but with an additional branching to avoid the CAS when we don't need it. Would that be an interesting tradeoff?